### PR TITLE
support build specific tag

### PR DESCRIPTION
**Pull request by onlymellb**
### What problem does this PR solve? 
Make tag build job also use CI script

### Check List 

Tests 

 - Unit test
 - E2E test

Code changes

 - Has CI related scripts change

### Does this PR introduce a user-facing change?:
 
 ```release-note

 ```

### CODE REVIEW

**Comment by onlymellb** [comment]
I have tested this script to build tag `v1.1.0-alpha.4`

File: ci/pingcap_tidb_operator_build_kind.groovy [hunk 1]
```diff
@@ -37,7 +37,7 @@ def call(BUILD_BRANCH, CREDENTIALS_ID, CODECOV_CREDENTIALS_ID) {
 							submoduleCfg: [],
 							userRemoteConfigs: [[
 								credentialsId: "${CREDENTIALS_ID}",
-								refspec: '+refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/*',
+								refspec: '+refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/* +refs/tags/*:refs/remotes/origin/tags/*',
```
Comment (cofyc) [inline ci/pingcap_tidb_operator_build_kind.groovy]: `+refs/tags/*:refs/remotes/origin/tags/*` is not needed. The first (also the default) refspec `+refs/heads/*:refs/remotes/origin/*` will fetch tags too.

```
# mkdir tidb-operator && cd tidb-operator 
# git init
# git config remote.origin.url git@github.com:pingcap/tidb-operator.git # timeout=10
# git fetch --tags --force --progress git@github.com:pingcap/tidb-operator.git +refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/* 
...
# git rev-parse v1.1.0-alpha.4^{commit}
77579e035ca92d578f779b482488d2a5a533c86c
```

Comment (onlymellb) [reply ci/pingcap_tidb_operator_build_kind.groovy]: Remove it

**Pull request merged**